passive/CookieHTTPOnly.js

```diff
@@ -1,37 +1,32 @@
 // Cookie HttpOnly Check by freakyclown@gmail.com
 
-function scan(ps, msg, src) {
-  var alertRisk = 1;
-  var alertConfidence = 2;
-  var alertTitle = "Cookie set without HTTPOnly Flag(script)";
-  var alertDesc =
-    "A cookie has been set without the HttpOnly flag, which means that the cookie can be accessed by JavaScript. If a malicious script can be run on this page then the cookie will be accessible and can be transmitted to another site. If this is a session cookie then session hijacking may be possible.";
-  var alertSolution = "Ensure that the HttpOnly flag is set for all cookies.";
+var ScanRuleMetadata = Java.type(
+  "org.zaproxy.addon.commonlib.scanrules.ScanRuleMetadata"
+);
 
-  var cweId = 0;
-  var wascId = 13;
-
-  var url = msg.getRequestHeader().getURI().toString();
-  var headers = msg.getResponseHeader().getHeaders("Set-Cookie");
+function getMetadata() {
+  return ScanRuleMetadata.fromYaml(`
+id: 100003
+name: Cookie Set Without HttpOnly Flag
+description: >
+  A cookie has been set without the HttpOnly flag, which means that the cookie can be accessed by JavaScript.
+  If a malicious script can be run on this page then the cookie will be accessible and can be transmitted to another site.
+  If this is a session cookie then session hijacking may be possible.
+solution: Ensure that the HttpOnly flag is set for all cookies.
+risk: low
+confidence: medium
+cweId: 0
+wascId: 13  # WASC-13: Information Leakage
+status: alpha
+`);
+}
 
-  if (headers != null) {
+function scan(helper, msg, src) {
+  var cookies = msg.getResponseHeader().getHeaders("Set-Cookie");
+  if (cookies != null) {
     var re_noflag = /([Hh][Tt][Tt][Pp][Oo][Nn][Ll][Yy])/g;
-    if (!re_noflag.test(headers)) {
-      ps.raiseAlert(
-        alertRisk,
-        alertConfidence,
-        alertTitle,
-        alertDesc,
-        url,
-        "",
-        "",
-        "",
-        alertSolution,
-        headers,
-        cweId,
-        wascId,
-        msg
-      );
+    if (!re_noflag.test(cookies)) {
+      helper.newAlert().setMessage(msg).raise();
```

Comment:
Is the evidence no longer being set because it's not an exact match in the response? (same for some of the following changes)

Reply:
No, that wasn't the intention, I missed setting it, good catch.